test/python/circuit/test_controlled_gate.py

```diff
@@ -70,6 +71,16 @@ def test_controlled_rz(self):
         theta = 0.5
         self.assertEqual(RZGate(theta).control(), CrzGate(theta))
 
+    def test_controlled_ry(self):
+        """Test creation of controlled ry gate"""
+        theta = 0.5
+        self.assertEqual(RYGate(theta).control(), CryGate(theta))
+
+    def test_controlled_rx(self):
```

Comment:
Since the `RXGate.control` just looks for the name and doesn't check the unrolling / matrices are equivalent I would suggest adding some matrix based tests that the definition is correct. Eg something like:

```python
def crx(theta):
    return qi.Operator(RXGate(theta)).tensor(qi.Operator.from_label('1')) + \
    qi.Operator.from_label('I').tensor(qi.Operator.from_label('0'))

np.alltrue([qi.Operator(CrxGate(theta)) == crx(theta) for theta in np.linspace(0, 2 * np.pi, 10)])
```

Reply:
Something like what @chriseclectic wrote may already be tested for Crx and Cry in `test_controlled_standard_gates`.

Reply:
Hi, thanks for your comment.
`test_unroller` tests for the correct unrolling of crx and cry. Do you think that would suffice? To my knowledge, there's also no direct matrix test for crz, for example.

Reply:
The suggested test checks that the controlled unitary explicitly matches the base gate. I don't believe this is done in test_unroller. I do believe it is tested in the function I mentioned however, along with Crz, so perhaps you don't need to add any test here.